File: tournaments/differentValuesInMultiplicationTable2/differentValuesInMultiplicationTable2.js
function differentValuesInMultiplicationTable2(n, m) {
    const store = new Set();

    for (let i = 1; m + 1 > i; ++i)
        for (let j = 1; n + 1 > j; ++j)
            store.add(i * j);

    return store.size;
}

File: tournaments/differentValuesInMultiplicationTable2/differentValuesInMultiplicationTable2.cpp
int differentValuesInMultiplicationTable2(int n, int m) {
  std::vector<bool> found(n * m + 1, false);
  int result = 0;

  for (int i = 1; i <= n; i++) {
    for (int j = 1; j <= m; j++) {
      found[i * j] = true;
    }
  }

  for (int value = 1; value <= n * m; value++) {
    if (found[value]) {
      result++;
    }
  }

  return result;
}
